SConstruct

```diff
@@ -27,6 +27,15 @@ active_platform_ids = []
 platform_exporters = []
 platform_apis = []
 
+# NOTE: It is safe to generate this file here, since this is still executed serially
+f = open("core/build_arguments.gen.h", "w")
```

Comment:
I think this could go in `main/SCsub` and output the file in `main`, it's not related to `core` as is.